contracts/linear/src/validator_pool.rs

```diff
@@ -646,21 +652,25 @@ impl LiquidStakingContract {
     }
 }
 
+/// How to add a new variant for VersionedValidator:
+/// 1. Put the current definition of Validator into legacy.rs as `ValidatorVx_x_x`
+/// 2. Update the current Validator struct
+/// 3. Implement `From<ValidatorVx_x_x> for Validator` so that we can migrate
+///    from the previous version of Validator to the latest.
+/// 4. Insert a new variant of VersionedValidator just BEFORE `Current(Validator)`.
+///    It stands for the previous version of Validator, which should be version
+///    that production is using at the time of writing the code.
+///    Due to the fact that Borsh use variant index instead of name to serialize,
+///    the new variant will be deserialzed to the previous version correctly after
+///    new code is deployed.
+/// 5. Update `impl From<VersionedValidator> for Validator`, to match the new variant.
```

Comment:
And also implement `From<ValidatorVx_x_x> for VersionedValidator` I think?

Reply:
this is optional I think? currently you don't need to convert from an old version of validator to `VersionedValidator`

Reply:
oh, yes. Then I think we can also remove the current implementation in `legacy.rs`.

```rs
impl From<ValidatorV1_0_0> for VersionedValidator {
    fn from(v: ValidatorV1_0_0) -> Self {
        VersionedValidator::V0(v)
    }
}
```

and

```rs
impl From<ValidatorV1_3_0> for VersionedValidator {
    fn from(v: ValidatorV1_3_0) -> Self {
        VersionedValidator::V1(v)
    }
}
```